src/App.tsx

```diff
@@ -1,15 +1,53 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
+import debounce from 'lodash.debounce';
 import { peopleFromServer } from './data/people';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [chosenPerson, setChosenPerson] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 300), []);
+
+  const onQueryChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (!query && !e.target.value.trim()) {
+      return;
+    }
+
+    setQuery(e.target.value);
+    if (e.target.value.trim() !== appliedQuery) {
+      applyQuery(e.target.value);
+      setChosenPerson('');
+    }
```

Comment:
The naming convention for the event handler function `onQueryChange` does not follow the conventions mentioned in the checklist. According to the conventions, handler names should start with 'handle' and the prop names which are used to pass these handlers down start with 'on'. So, `onQueryChange` should be renamed to `handleQueryChange`.